product_pricelist_rules/models/product_pricelist.py

```diff
@@ -123,8 +123,11 @@ def domain_by_pricelist(self, pricelist_id, product_id=False,
                 tmpl_id = product_tmpl_id
                 if not tmpl_id and product_id:
                     tmpl_id = product_obj.browse(product_id).product_tmpl_id.id
-                if not suppinfo_obj.search([('product_tmpl_id', '=', tmpl_id),
-                                            ('name', '=', partner_id)]):
+                suppinfo_lst = suppinfo_obj.search(
+                    [('product_tmpl_id', '=', tmpl_id),
+                     ('name', '=', partner_id)])
+                if (not suppinfo_lst or not
```

Comment:
No necesitas hacer la doble comparación: `if not suppinfos.mapped('pricelist_ids'):` te devuelve lo mismo.